nixos/modules/services/misc/spice-webdavd.nix

```diff
@@ -0,0 +1,34 @@
+{ config, pkgs, lib, ... }:
+
+with lib;
+let
+  cfg = config.services.spice-webdavd;
+in
+{
+  options = {
+    services.spice-webdavd = {
+      enable = mkEnableOption "Spice guest webdav proxy daemon";
+    };
+  };
+
+  config = mkIf cfg.enable {
+    # pretty useless without these
+    services.davfs2.enable = true;
+    services.avahi.enable = true;
+
+    # udev rule to proxy when it sees the SPICE socket
+    services.udev.packages = [ pkgs.phodav ];
+
+    systemd.services.spice-webdavd = {
+      description = "spice-webdav proxy daemon";
+      requires = [ "dbus.service" "avahi-daemon.service" ];
+      after = [ "dbus.service" "avahi-daemon.service" ];
+
+      serviceConfig = {
+        Type = "simple";
+        ExecStart = "${pkgs.phodav}/bin/spice-webdavd -p 9843";
```

Comment:
Don't directly reference a package, use a [package option](https://github.com/nixos/nixpkgs/blob/d8e52b5b53cf58babb1f48c9e7e28ab0a198a219/nixos/modules/services/games/minecraft-server.nix#L153). 

(i write this after having to add that to [tw](https://github.com/nixos/nixpkgs/commit/5e1bb704103)o [modules](https://github.com/nixos/nixpkgs/commit/6f7b6952325) recently so I could [patc](https://github.com/ckiee/nixfiles/blob/62c5cc485c2310e6b13daa76151651fa1748cab3/modules/services/gitd/default.nix#L115)h [them](https://github.com/ckiee/nixfiles/blob/62c5cc485c2310e6b13daa76151651fa1748cab3/modules/services/gitd/default.nix#L7))

Reply:
Added.